sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -712,6 +711,51 @@ object PushPredicateThroughAggregate extends Rule[LogicalPlan] with PredicateHel
 }
 
 /**
+  * Reorder the joins so that the bottom ones have at least one condition.
+  */
```

Comment:
Can you add to this comment what makes this rule stable? It's not obvious from reading the code.